lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2711,19 +2830,31 @@ ParseResult FIRCircuitParser::parseCircuit() {
 //===----------------------------------------------------------------------===//
 
 // Parse the specified .fir file into the specified MLIR context.
-OwningModuleRef circt::firrtl::importFIRRTL(SourceMgr &sourceMgr,
-                                            MLIRContext *context,
-                                            FIRParserOptions options) {
+OwningModuleRef
+circt::firrtl::importFIRRTL(SourceMgr &sourceMgr,
+                            llvm::Optional<llvm::json::Value> annotations,
+                            MLIRContext *context, FIRParserOptions options) {
   auto sourceBuf = sourceMgr.getMemoryBuffer(sourceMgr.getMainFileID());
 
+  llvm::json::Path::Root root;
+  llvm::StringMap<llvm::SmallVector<Annotation>> annotationMap;
+  if (annotations) {
+    for (auto a : *annotations.getValue().getAsArray()) {
+      if (failed(fromJSON(a, annotationMap, root, context))) {
+        llvm::errs() << "Failed to parse annotation file...\n";
```

Comment:
Please use emitError on a location in the fir file.

Reply:
Fixed in squash commits.